src/status_im2/contexts/chat/events.cljs

```diff
@@ -103,18 +103,18 @@
           :chats-home-list #{}
           :removed-chats   #{}}
          (map (map-chats cofx) chats))]
-    (rf/merge
-     cofx
-     (merge {:db (-> db
-                     (update :chats merge all-chats)
-                     (update :chats-home-list set/union chats-home-list)
-                     (update :chats #(apply dissoc % removed-chats))
-                     (update :chats-home-list set/difference removed-chats))}
-            (when (not-empty removed-chats)
-              {:clear-message-notifications
-               [removed-chats
-                (get-in db [:profile/profile :remote-push-notifications-enabled?])]}))
-     leave-removed-chat)))
+    {:db (-> db
+             (update :chats merge all-chats)
+             (update :chats-home-list set/union chats-home-list)
+             (update :chats #(apply dissoc % removed-chats))
+             (update :chats-home-list set/difference removed-chats))
+     :fx [(when (not-empty removed-chats)
+            [:clear-message-notifications
+             [removed-chats
```

Comment:
oh idk why, but it's so confusing, effect mixed with events, brrrr

```[:clear-message-notifications] and [:chat/leave-removed-chat]``` in da mix, one is effect and another is event

Reply:
Understandable, it's only because you're used to the current way for a few years @flexsurfer. For many others, this is as clean as it gets.